### PR TITLE
Fix nthreads on >=1.9. Update and fix CI

### LINKED ISSUE #1: "failed to start primary task" with Julia 1.9 and nthreads(:interactive) > 0
Mirroring issue from: https://github.com/JuliaFolds/FoldsThreads.jl/issues/21

> Reproducible on Julia 1.9 with:
> 
> ```
> julia --threads 2,1 -e "using FoldsThreads"  
> ┌ Error: failed to start primary task
> └ @ FoldsThreads.Implementations ~/.julia/packages/FoldsThreads/aWnRU/src/root_spawners.jl:34
> ERROR: InitError: TaskFailedException
> 
>     nested task error: BoundsError: attempt to access 2-element Vector{Union{Nothing, Channel{Any}}} at index [3]
>     Stacktrace:
>      [1] setindex!
>        @ ./array.jl:969 [inlined]
>      [2] (::FoldsThreads.Implementations.var"#29#31")()
>        @ FoldsThreads.Implementations ~/.julia/packages/FoldsThreads/aWnRU/src/root_spawners.jl:62
>      [3] macro expansion
>        @ ~/.julia/packages/FoldsThreads/aWnRU/src/utils.jl:6 [inlined]
> ...
> ```
> 
> It's caused by the assumption that the threadids are 1:nthreads, which is in case of nthreads(:interactive) > 0 not valid anymore:

> https://github.com/JuliaFolds/FoldsThreads.jl/blob/6bc5651bbbfac60df4bc50ed84b4de2a9f9817a2/src/root_spawners.jl#L58-L64

> EACH_THREAD is too short since the ids are distributed like this:
> `[1:nthreads(:interactive); nthreads(:interactive) .+ (1:nthreads())]`

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes #1  (original https://github.com/JuliaFolds/FoldsThreads.jl/issues/21)

[It was suggested by @moelf](https://github.com/JuliaFolds/FoldsThreads.jl/issues/21#issuecomment-1572594958) to also add an `init!` function to update if the threading has changed, but can that come in a following PR because I'd rather get this fix out promptly?

```
% julia -t2,1 --project=/Users/ian/Documents/GitHub/FoldsThreads.jl
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.1 (2023-06-07)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using FoldsThreads
[ Info: Precompiling FoldsThreads [9c68100b-dfe1-47cf-94c8-95104e173443]

julia> 
```

cc. @MasonProtter @moelf @MarkusSchildhauer

### CODE REVIEW

**Comment by MasonProtter** [comment]
Ah very cool, that was an easier fix than I had worried it might be!

**Comment by MasonProtter** [comment]
So this package has some somewhat arcane test suite stuff using Run.jl and that's causing you problems here with the test failing on nightly and 1.9. Basically it's checking out specific manifests provided in the test directory.. 

The solution we've used in some other Folds repos is to just disable the Run.jl stuff in the `.github/workflows/test.yml` file and use the regular test infrastructure instead.

File: Project.toml [hunk 1]
```diff
@@ -17,9 +17,14 @@ InitialValues = "0.2.10, 0.3"
 SplittablesBase = "0.1.11"
 Transducers = "0.4.55"
 julia = "1"
+Folds = "=0.2.6"
+FLoops = "=0.1.11"
```
Comment (MasonProtter) [inline Project.toml]: hm, current Floops is v0.2.1, was it installing that version before and causing problems?

Comment (IanButterworth) [reply Project.toml]: Seems so

Comment (IanButterworth) [reply Project.toml]: Actually, no. These were the versions when it was failing
```
  [7d9f7c33] Accessors v0.1.32
  [e30172f5] Documenter v0.27.24
⌅ [cc61a311] FLoops v0.1.12
  [41a02a25] Folds v0.2.8
  [9c68100b] FoldsThreads v0.1.2-DEV `~/work/FoldsThreads.jl/FoldsThreads.jl`
  [069b7b12] FunctionWrappers v1.1.3
  [22cec73e] InitialValues v0.3.1
  [171d559e] SplittablesBase v0.1.15
  [28d57a85] Transducers v0.4.76
  [8dfed614] Test `@stdlib/Test`
```

Comment (IanButterworth) [reply Project.toml]: Which is worrying that a breakage happened in a patch version of Folds or FLoops

Comment (IanButterworth) [reply Project.toml]: Looks like it's a breakage in Folds `0.2.6` -> `0.2.8`

**Comment by IanButterworth** [comment]
Ok, I think this is ready for review.

I opened #3. Perhaps that's for a follow-on PR?

**Comment by MasonProtter** [comment]
> I opened #3. Perhaps that's for a follow-on PR?

yeah that can be handled separately.

**Review by MasonProtter — COMMENTED** [review]
All these changes make sense, but one thing I'm not totally sure on is how many threads the CI is running with. 

Maybe we should be explicitly making single threaded CI jobs, multithreaded CI jobs, and on v1.9+, jobs that have interactive *and* regular threads?

**Comment by MasonProtter** [comment]
Yay :tada:! Thank you @IanButterworth for tackling this.

**Comment by IanButterworth** [comment]
Nice. Note that General still points to `JuliaFolds` for this package. We should probably find-and-replace all in General?

**Comment by MasonProtter** [comment]
Yeah, just pending this one: https://github.com/JuliaRegistries/General/pull/85601

**Comment by MarkusSchildhauer** [comment]
Many Thanks to @IanButterworth for solving this!

**Pull request merged**